gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -521,6 +522,24 @@ protected int getDefaultPort() {
     return result;
   }
 
+  public boolean hasPathCapability(Path path, String capability) throws IOException {
+    switch (validatePathCapabilityArgs(path, capability)) {
+      // TODO: remove string literals in favor of Constants in CommonPathCapabilities.java
+      // from Hadoop 3 when Hadoop 2 is no longer supported
+      case "fs.capability.paths.append":
+      case "fs.capability.paths.concat":
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static String validatePathCapabilityArgs(Path path, String capability) {
```

Comment:
Make it private, if it's used only internally

Reply:
we cannot make it public, since the Interface exposes it as `public`

Reply:
But this is a static method that is not part of the interface? So probably we can make it private?

Reply:
This is part of the [interface](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/FileSystem.html#hasPathCapability-org.apache.hadoop.fs.Path-java.lang.String-)

The build breaks if we make it private

Reply:
my bad, was checking the other method. making the method private